src/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -61,104 +61,138 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
 
 	@Override
 	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		String siteIdentifier = getSiteIdentifier(msg);
+
+		visitedSiteIdentifiers.add(siteIdentifier);

```

Comment:
We need to return if it was already present, also we can change `visitedSiteIdentifiers` to a `Set` (which returns if the `siteIdentifier` was added or not).